exonum-java-binding/core/rust/integration_tests/tests/service_runtime.rs

```diff
@@ -110,32 +140,13 @@ fn create_service_for_unknown_artifact() {
     runtime.create_service("unknown:artifact:id");
 }
 
-fn create_runtime() -> JavaServiceRuntime {
-    let jvm_config = JvmConfig {
-        args_prepend: vec![],
-        args_append: vec![],
-        jvm_debug_socket: None,
-    };
-
-    let runtime_config = RuntimeConfig {
-        log_config_path: "".to_owned(),
-        port: 6300,
-    };
-
-    let config = Config {
-        jvm_config,
-        runtime_config,
-    };
-
-    let internal_config = InternalConfig {
-        system_class_path: get_fakes_classpath(),
-        system_lib_path: None,
-    };
-
-    JavaServiceRuntime::new(config, internal_config)
+// Creates a new instance of JavaServiceRuntime for same JVM.
+fn get_runtime() -> JavaServiceRuntime {
+    JavaServiceRuntime::new(VM.clone(), get_port())
 }
 
-// Returns a guard to the service runtime that helps executing tests in turn.
-fn get_runtime() -> MutexGuard<'static, JavaServiceRuntime> {
-    SERVICE_RUNTIME.lock()
+// Provides port number that is unique within the scope of tests module.
+fn get_port() -> i32 {
+    static PORT: AtomicUsize = ATOMIC_USIZE_INIT;
+    PORT.fetch_add(1, Ordering::SeqCst) as i32 + 6300
```

Comment:
Why are these complications? Can't we just pass 0 so that the server picks any available port?

Reply:
Done, thanks for the idea, somehow I forgot it's possible to do it that way...